pkg/gofr/subscriber_test.go

```diff
@@ -72,10 +79,12 @@ func TestSubscriptionManager_HandlerError(t *testing.T) {
 
 		// Run the subscriber in a goroutine
 		go func() {
-			subscriptionManager.startSubscriber("test-topic",
+			err := subscriptionManager.startSubscriber(context.Background(), "test-topic",
 				func(*Context) error {
 					return handleError("error in test-topic")
 				})
+
+			assert.Contains(t, handleError("error in test-topic").Error(), err.Error())
```

Comment:
```suggestion
			assert.ErrorContains(t, err,  handleError("error in test-topic").Error())
```

or 

```suggestion
			assert.ErrorContains(t, err, "error in test-topic")
```

or 


```suggestion
			assert.ErrorIs(t, err, handleError("error in test-topic"))
```

Maybe you should return a sentinel and use

```suggestion
      assert.ErrorIs(t, err, ErrSentinel)
			assert.ErrorContains(t, err, "error in test-topic")
```

Reply:
Changed